packages/ra-tree-ui-materialui/src/TreeListToolbar.tsx

```diff
@@ -0,0 +1,44 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import Toolbar from '@material-ui/core/Toolbar';
+import { withStyles, createStyles } from '@material-ui/core/styles';
+
+const styles = createStyles({
+    toolbar: {
+        justifyContent: 'space-between',
+    },
+});
+
+const TreeListToolbar = ({
+    classes,
+    filters,
+    permanentFilter, // set in the List component by the developer
+    actions,
+    exporter,
+    ...rest
+}) => (
+    <Toolbar className={classes.toolbar}>
+        <span />
```

Comment:
I understand you're using ` justifyContent: 'space-between'` and this `span` to push the actions to the right. But that means that 90% of the line is the toolbar... Wrong solution IMHO.

Reply:
That's how it is done in the ListToolbar from which I just removed the filters